### PR TITLE
Monitor resque

### PR DESCRIPTION
**Pull request by tallenaz**
check whether queues are backed up
  and for failed jobs

### CODE REVIEW

File: config/initializers/okcomputer.rb [hunk 1]
```diff
@@ -59,6 +60,16 @@ def check
 
 OkComputer::Registry.register 'ruby_version', OkComputer::RubyVersionCheck.new
 
+# check for backed up resque queues
+Resque.queues.each do |queue|
+  OkComputer::Registry.register "feature-backed-up-#{queue}",
+                                OkComputer::ResqueBackedUpCheck.new(queue, 5_000_000)
+end
+
+# check for failed resque jobs
```
Comment (ndushay) [inline config/initializers/okcomputer.rb]: Can we / should we do this per error queue instead?

If we keep this, what do you think of as "feature-resque-failures-threshold" as the name?

File: config/initializers/okcomputer.rb [hunk 2]
```diff
@@ -59,6 +60,16 @@ def check
 
 OkComputer::Registry.register 'ruby_version', OkComputer::RubyVersionCheck.new
 
+# check for backed up resque queues
+Resque.queues.each do |queue|
+  OkComputer::Registry.register "feature-backed-up-#{queue}",
```
Comment (ndushay) [inline config/initializers/okcomputer.rb]: what do you think of "feature-#{queue}-queue-depth" as a name?

File: config/initializers/okcomputer.rb (hunk 1 above)
Comment (tallenaz) [reply config/initializers/okcomputer.rb]: The native OKComputer Resque tools don't have the ability to monitor per error queue, just the total number of errors. 

I'm willing to look into writing custom checks, if we think there's enough of a need.

+1 to the name.

File: config/initializers/okcomputer.rb (hunk 2 above)
Comment (tallenaz) [reply config/initializers/okcomputer.rb]: Sounds good.

**Comment by tallenaz** [comment]
I timeboxed 30 minutes looking into how to monitor an individual error queue, and it seems like a not-easy problem.

It's easy to get queue names with `Resque.queues`, and easy to get an individual error queue for a queue name with `Resque::Failure.failure_queue_name(Resque.queues[0])`, for example. But I don't see a straightforward way to get the count of a particular failure queue.

**Comment by ndushay** [comment]
Hrmm - can you add that to storytime?  Having one check for failures for all queues seems sub-optimal.  I'm merging this but I think we're going to need to discuss ...

**Pull request merged**